### ISSUE TITLE
Decompile apps further into Java code

### ISSUE THREAD

**Issue by Nik-Kot** [apk-editor/APK-Explorer-Editor#33]
Add support for decompiling apps into Java code using jadx as a library.
https://github.com/skylot/jadx/wiki/Use-jadx-as-a-library

I am willing to contribute of this idea gets approval from creators.

**Comment by apk-editor** [comment]
Hi @Nik-Kot 
Thanks for the suggestion. However, do they really convert a dex file to java code or just a smali file (AEE already do it)? If the first one is the case, I'm really interested and looking forward to your contributions.
Thanks again

**Comment by Nik-Kot** [comment]
> Hi @Nik-Kot Thanks for the suggestion. However, do they really convert a dex file to java code or just a smali file (AEE already do it)? If the first one is the case, I'm really interested and looking forward to your contributions. Thanks again

.dex into Java code. You can download jadx for PC and try it out, there are also many other features.

**Comment by Nik-Kot** [comment]
Btw their license is Apache 2.0, is this ok with you?

**Comment by apk-editor** [comment]
> Btw their license is Apache 2.0, is this ok with you? We only need to use their ready library as a dependency.

That will be fine (to the best of my knowledge).

**Comment by Nik-Kot** [comment]
Also jadx supports decompilation of .apk, .jar, .class, .smali, .aar, .arsc, and .aab.
What would be the best way for you if I start working on this? Should I make private/public fork or do something else?

**Comment by apk-editor** [comment]
> Also jadx supports decompilation of .apk, .jar, .class, .smali, .aar, .arsc, and .aab. What would be the best way for you if I start working on this? Should I make private/public fork or do it some other way?

Ideally, a PR from a publically available fork?

**Comment by Nik-Kot** [comment]
> > Also jadx supports decompilation of .apk, .jar, .class, .smali, .aar, .arsc, and .aab. What would be the best way for you if I start working on this? Should I make private/public fork or do it some other way?
> 
> Ideally, a PR from a publically available fork?

There you go
https://github.com/Nik-Kot/APK-Explorer-Editor
Also I invited you you to collaborators, you can make changes if you wish too. I will need some time to get acquainted with your code, don't expect anything right away.

**Comment by Nik-Kot** [comment]
> Thanks. And take you time. I am also busy and spending minimum time for apps these days.

Np. I will update you on the progress here.

**Comment by apk-editor** [comment]
> Hm, a question. How does jadx comply with Google Play guidelines? Should I restrict it's usage only to "full" version or not?

Not sure at this moment. But, feel free to work on it for both flavours. If needed, I'll take care of the rest.

**Comment by apk-editor** [comment]
@Nik-Kot 
I think we will face a serious issue here. As per the developer(s) of jadx, the library may not 100% decompile the classes (please see [ReadMe](https://github.com/skylot/jadx/blob/master/README.md))) so that rebuilding will be difficult. It will still be an interesting tool for those who just want to see the code.

**Comment by Nik-Kot** [comment]
> @Nik-Kot I think we will face a serious issue here. As per the developer(s) of jadx, the library may not 100% decompile the classes (please see [ReadMe](https://github.com/skylot/jadx/blob/master/README.md))) so that rebuilding will be difficult. It will still be an interesting tool for those who just want to see the code.

Yes, I am well aware of that. My plan is to make usege of JADX completely optional and applicable to single class files.

**Comment by thewanderingtraderm** [comment]
@apk-editor @Nik-Kot when is this gonna be added?
